### PR TITLE
update to version 3.16.5

### PR DESCRIPTION
**Pull request by matthiasdiener**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by matthiasdiener** [comment]
This is ready for review @conda-forge/petsc

**Comment by minrk** [comment]
Where did you determine that run_test.sh is unused? It should be [run by conda-build](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#run-test-script). If it isn't running for some reason, those tests still should be run. The fix isn't to remove the tests.

File: recipe/build.sh [hunk 1]
```diff
@@ -96,35 +96,24 @@ for f in $PETSC_ARCH/lib/petsc/conf/petscvariables $PETSC_ARCH/lib/pkgconfig/PET
...
 
 make
 
 
 if [[ "$CONDA_BUILD_CROSS_COMPILATION" != "1" ]]; then
-  # FIXME: Workaround mpiexec setting O_NONBLOCK in std{in|out|err}
-  # See https://github.com/conda-forge/conda-smithy/pull/337
-  # See https://github.com/pmodels/mpich/pull/2755
-  make check MPIEXEC="${RECIPE_DIR}/mpiexec.sh"
+  make check
```
Comment (dalcinl) [inline recipe/build.sh:116]: @mirk Perhaps we should try MPICH again for the infamous issue that forced us to use this mpiexec hack.

Comment (matthiasdiener) [reply recipe/build.sh:116]: The fix seems to have been merged to mpich, is mpiexec.sh still necessary?

**Comment by matthiasdiener** [comment]
> Where did you determine that run_test.sh is unused? It should be [run by conda-build](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#run-test-script). If it isn't running for some reason, those tests still should be run. The fix isn't to remove the tests.

My apologies, I should have looked more carefully.

File: recipe/build.sh [hunk 2]
```diff
@@ -96,35 +96,24 @@ for f in $PETSC_ARCH/lib/petsc/conf/petscvariables $PETSC_ARCH/lib/pkgconfig/PET
...
-  fi
-}
-
 # Remove abspath of ${BUILD_PREFIX}/bin/python
-sedinplace "s%${BUILD_PREFIX}/bin/python%python%g" $PETSC_ARCH/include/petscconf.h
-sedinplace "s%${BUILD_PREFIX}/bin/python%python%g" $PETSC_ARCH/lib/petsc/conf/petscvariables
-sedinplace "s%${BUILD_PREFIX}/bin/python%/usr/bin/env python%g" $PETSC_ARCH/lib/petsc/conf/reconfigure-arch-conda-c-opt.py
+sed -i.bak "s%${BUILD_PREFIX}/bin/python%python%g" $PETSC_ARCH/include/petscconf.h
+sed -i.bak "s%${BUILD_PREFIX}/bin/python%python%g" $PETSC_ARCH/lib/petsc/conf/petscvariables
+sed -i.bak "s%${BUILD_PREFIX}/bin/python%/usr/bin/env python%g" $PETSC_ARCH/lib/petsc/conf/reconfigure-arch-conda-c-opt.py
```
[the remark this answers is not in the document]
Comment (matthiasdiener) [reply recipe/build.sh]: I restored sedinplace.

**Comment by dalcinl** [comment]
You have 9 commits with lots of reverts just to change a few lines in a few lines. Can you please fixup/squash your commits? Git history noise hinders maintenance.

**Comment by matthiasdiener** [comment]
> You have 9 commits with lots of reverts just to change a few lines in a few lines. Can you please fixup/squash your commits? Git history noise hinders maintenance.

Good point, I just squashed the commits.

**Pull request merged**